drivers/storage/partmgr/partition.c

```diff
@@ -173,6 +172,83 @@ PartitionHandleStartDevice(
     return STATUS_SUCCESS;
 }
 
+/**
+ * @brief
+ * Notifies MountMgr to delete all mount points
+ * associated with the given volume.
+ *
+ * @note    This should belong to volmgr.sys and act on a PVOLUME_EXTENSION.
+ **/
+static
+CODE_SEG("PAGE")
+NTSTATUS
+VolumeDeleteMountPoints(
+    _In_ PPARTITION_EXTENSION PartExt)
+{
+    NTSTATUS Status;
+    ULONG InputBufferLength;
+    UNICODE_STRING MountMgr;
+    PUNICODE_STRING DeviceName;
+    PDEVICE_OBJECT DeviceObject;
+    PFILE_OBJECT FileObject = NULL;
+    PMOUNTMGR_MOUNT_POINT InputBuffer = NULL;
+    PMOUNTMGR_MOUNT_POINTS OutputBuffer = NULL;
+
+    /* Setup the volume device name for deleting its mount points */
+    DeviceName = &PartExt->DeviceName;
+
+    /* Allocate the input buffer for MountMgr */
+    InputBufferLength = sizeof(MOUNTMGR_MOUNT_POINT) + DeviceName->Length;
+    InputBuffer = ExAllocatePoolZero(PagedPool, InputBufferLength, TAG_PARTMGR);
+    if (!InputBuffer)
+        return STATUS_INSUFFICIENT_RESOURCES;
```

Comment:
This avoids redundant zeroing of allocated memory (note the RtlCopyMemory below):
```suggestion
    InputBuffer = ExAllocatePoolWithTag(PagedPool, InputBufferLength, TAG_PARTMGR);
    if (!InputBuffer)
        return STATUS_INSUFFICIENT_RESOURCES;
        
    RtlZeroMemory(InputBuffer, sizeof(*InputBuffer));
```

Reply:
addressed